### DISCUSSION TITLE
How do I install and use cpr?

### DISCUSSION THREAD

**Discussion by elibroftw** [microsoft/vcpkg#21993] [Q&A] [answered]
What is the streamlined process for installing cpr? 
I did 
- `vcpkg install cpr`
 - It told me
 ```bash
 The package cpr provides CMake targets:

    find_package(cpr CONFIG REQUIRED)
    target_link_libraries(main PRIVATE cpr::cpr)
 ```
- I created a cmake project in Visual Studio 2022
- I set the toolchain in `CMakeSettings.json`
- I added the package lines to `CMakeLists.txt`
- Now I get this error:
```
Severity	Code	Description	Project	File	Line	Suppression State
Error		CMake Error at C:/Users/maste/Documents/GitHub/vcpkg/scripts/buildsystems/vcpkg.cmake:788 (_find_package):
  Could not find a package configuration file provided by "cpr" with any of
  the following names:

    cprConfig.cmake
    cpr-config.cmake

  Add the installation prefix of "cpr" to CMAKE_PREFIX_PATH or set "cpr_DIR"
  to a directory containing one of the above files.  If "cpr" provides a
  separate development package or SDK, be sure it has been installed.		C:/Users/maste/Documents/GitHub/vcpkg/scripts/buildsystems/vcpkg.cmake	788	
```

What am I missing?
My `CMakeList.txt` for reference
```cmake
# CMakeList.txt : CMake project for CMakeProject1, include source and define
# project specific logic here.
#
cmake_minimum_required (VERSION 3.8)

project ("CMakeProject1")

# Add source to this project's executable.
add_executable (CMakeProject1 "CMakeProject1.cpp" "CMakeProject1.h")

find_package(cpr CONFIG REQUIRED)
target_link_libraries(CMakeProject1 PRIVATE cpr::cpr)
```

**Comment by elibroftw** [comment, marked as answer]
I deleted the contents of the build folder and it works now. Will uninstall cpr and start from scratch again.

**Reply by CaejV** [reply]
I'm following your guide on youtube and cannot resolve this issue. Do you happen to know what steps you followed to resolve?

**Reply by elibroftw** [reply]
I think it has to do with caching which is why I deleted the build folder (either of my project or vcpkg's build folder)??? This was over 2 years ago and I didn't really write projects in C++ vcpkg after the video. It could do with the cmake cache not including the vcpkg toolchain file? The video you watched was many tries. Try following the blog post https://blog.elijahlopez.ca/posts/vcpkg-cmake-tutorial/

**Reply by CaejV** [reply]
thank you! I tried deleting cache prior and had no luck but for some reason this one time seemed to work. Thank you so much!
